### PR TITLE
Huffman decoder will grow output buffer, if you want it to

### PR DESCRIPTION
**Pull request by graebm**
Add option to grow output buffer while decoding.

The alternative is to always have the output buffer be big enough for the largest string you allow, but that's probably a waste of space in the normal case.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by graebm** [comment]
not ready to merge this yet...

File: source/huffman.c [hunk 1]
```diff
@@ -270,8 +270,15 @@ int aws_huffman_decode(
         }
 
         if (output->len == output->capacity) {
-            /* Check if we've hit the end of the output buffer */
-            return aws_raise_error(AWS_ERROR_SHORT_BUFFER);
+            /* Check if we've hit the end of the output buffer.
+             * Grow buffer, or raise error, depending on settings */
+            if (decoder->allow_growth) {
+                if (aws_byte_buf_reserve_relative(output, output->capacity)) {
```
Comment (bretambrose) [inline source/huffman.c]: So policy is just double the size?

File: tests/huffman_test.c [hunk 2]
```diff
@@ -340,6 +340,28 @@ static int test_huffman_decoder_partial_output(struct aws_allocator *allocator,
     return AWS_OP_SUCCESS;
 }
 
+AWS_TEST_CASE(huffman_decoder_allow_growth, test_huffman_decoder_allow_growth)
```
Comment (bretambrose) [inline tests/huffman_test.c]: Is there already a test case that verifies failure when growth is disabled?

File: source/huffman.c (hunk 1 above)
Comment (graebm) [reply source/huffman.c]: yes, I'll add a comment

File: tests/huffman_test.c (hunk 2 above)
Comment (graebm) [reply tests/huffman_test.c]: yes, the previous test in the file `huffman_decoder_partial_output` tests the growth-disabled behavior

**Pull request merged**